vision_datasets/dataset/vision_dataset.py

```diff
@@ -0,0 +1,176 @@
+import copy
+import logging
+import os.path
+import pathlib
+import typing
+
+from PIL import Image, JpegImagePlugin
+from tqdm import tqdm
+
+from ..common import DatasetTypes
+from ..data_manifest import DatasetManifest, ImageDataManifest, ImageLabelWithCategoryManifest
+from ..data_reader import FileReader, PILImageLoader
+from ..dataset.base_dataset import BaseDatasetInfo
+from .base_dataset import BaseDataset
+
+logger = logging.getLogger(__name__)
+
+
+class VisionDataset(BaseDataset):
+    """Dataset class that accesses data from dataset manifest
+    """
+
+    def __init__(self, dataset_info: BaseDatasetInfo, dataset_manifest: DatasetManifest, coordinates='relative', dataset_resources=None):
+        """
+
+        Args:
+            dataset_info (BaseDatasetInfo): dataset info, containing high level information about the dataset, such as name, type, description, etc
+            dataset_manifest (DatasetManifest): dataset manifest containing meta data such as image paths, annotations, etc
+            coordinates (str): 'relative' or 'absolute', indicating the desired format of the bboxes returned.
+            dataset_resources (str): disposable resources associated with this dataset
+        """
+
+        assert dataset_manifest is not None
+        assert coordinates in ['relative', 'absolute']
+
+        super().__init__(dataset_info)
+
+        self.dataset_manifest = dataset_manifest
+        self.coordinates = coordinates
+        self._file_reader = FileReader()
+        self.dataset_resources = dataset_resources
+
+    @property
+    def labels(self):
+        return self.dataset_manifest.categories
+
+    def __len__(self):
+        return len(self.dataset_manifest.images)
+
+    def _get_single_item(self, index):
+        image_manifest: ImageDataManifest = self.dataset_manifest.images[index]
+        image = self._load_image(image_manifest.img_path)
+        target = image_manifest.labels
+        if self.coordinates == 'relative':
+            w, h = image.size
+            target = VisionDataset._convert_box_to_relative(image_manifest.labels, w, h, self.dataset_info)
+
+        return image, target, str(index)
+
+    def close(self):
+        self._file_reader.close()
+
+    def _load_image(self, filepath):
+        full_path = filepath.replace('\\', '/')
```

Comment:
it would be great if we could forget windows filepaths in image manifest. If we support windows, we need to test image paths like `C:\Users\`